Monitoring dev mode ARC in dev ui crash quarkus "dev flow"

### Describe the bug

In DEV-UI, in the configuration tab, activating the property `quarkus.arc.dev-mode.monitoring-enabled` implies a crash.
dev mode must be restarted.

```
2023-05-05 07:10:13,088 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 15 more
Caused by: java.lang.NoSuchMethodError: io.quarkus.vertx.http.runtime.CurrentVertxRequest_Bean: method 'void <init>()' not found
	at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans1(Unknown Source)
	at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Unknown Source)
	at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:120)
	at io.quarkus.arc.Arc.initialize(Arc.java:45)
	at io.quarkus.arc.runtime.ArcRecorder.initContainer(ArcRecorder.java:47)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources844392269.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.ArcProcessor$generateResources844392269.deploy(Unknown Source)
	... 16 more
```

### Expected behavior

If this property cannot be updated at runtime, it would be described in documentation and mentioned to the user in DEV UI.

### Actual behavior

_No response_

### How to Reproduce?


1. Start the quarkus dev mode
2. Go to dev ui 
3. Go to configuration tab
4. Activate the property `quarkus.arc.dev-mode.monitoring-enabled`. A notification appears to acknowlegde the update.

<img width="744" alt="quarkus arc dev-mode monitoring-enabled 2023-05-05" src="https://user-images.githubusercontent.com/42130250/236382751-27ed0df6-c65c-41ef-9387-50211ee079cc.png">

  The quarkus console indicates that reload is done

  `INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) Live reload total time: 0.173s`

5. Try to GET /hello or any other endpoints and you will see the stacktrace mentioned before.









### Output of `uname -a` or `ver`

Darwin Kernel Version 21.6.0

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 OpenJDK Runtime Environment (build 17.0.1+12-39) OpenJDK 64-Bit Server VM (build 17.0.1+12-39, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: /Users/fifcan/.m2/wrapper/dists/apache-maven-3.8.8-bin/67c30f74/apache-maven-3.8.8 Java version: 17.0.1, vendor: Oracle Corporation, runtime: /Users/fifcan/Library/Java/JavaVirtualMachines/openjdk-17.0.1/Contents/Home Default locale: fr_FR, platform encoding: UTF-8 OS name: "mac os x", version: "12.6.5", arch: "aarch64", family: "mac"

### Additional information

_No response_

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcDevModeConfig.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcDevModeConfig.java
@@ -8,6 +8,8 @@ public class ArcDevModeConfig {
 
     /**
      * If set to true then the container monitors business method invocations and fired events during the development mode.
+     * <p>
+     * NOTE: This config property should not be changed in the development mode as it requires a full rebuild of the application
      */
     @ConfigItem(defaultValue = "false")
     public boolean monitoringEnabled;

